dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,147 @@
+import logging
+from datetime import datetime
+
+from analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+from analyzer_helper.discord.utils.is_user_bot import UserBotChecker
+
+
+class DiscordTransformRawData(TransformRawDataBase):
+
+    def __init__(self):
+        """
+        Initializes the class with a bot checker.
+        """
+        self.user_bot_checker = UserBotChecker()
+
+    def create_interaction(self, interaction_type, users_engaged_id, interaction_role):
```

Comment:
Please annotate the parameters in the function header. It could be something like below ( my example is incomplete)

```
def create_interaction(self, interaction_type: str, users_engaged_id: list[str], 
```

Reply:
Also, the `interaction_role` might not be a very appropriate name (low understandability for devs). Please consider updating it to `type`